l10n_it_fatturapa_pec/models/mail_thread.py

```diff
@@ -110,56 +111,34 @@ def message_route(self, message, message_dict, model=None, thread_id=None,
                             message_dict)
-                else:
-                    _logger.info(
-                        'Can\'t route PEC E-Mail with Message-Id: {}'.format(
-                            message.get('Message-Id'))
-                    )
-                    if fetchmail_server.e_inv_notify_partner_ids:
-                        self.env['mail.mail'].create({
-                            'subject': _(
-                                "PEC message [%s] not processed"
-                            ) % message.get('Subject'),
-                            'body_html': _(
-                                "<p>"
-                                "PEC message with Message-Id %s has been read "
-                                "but not processed, as not related to an "
-                                "e-invoice.</p>"
-                                "<p>Please check PEC mailbox %s, at server %s,"
-                                " with user %s</p>"
-                            ) % (
-                                message.get('Message-Id'),
-                                fetchmail_server.name, fetchmail_server.server,
-                                fetchmail_server.user
-                            ),
-                            'recipient_ids': [(
-                                6, 0,
-                                fetchmail_server.e_inv_notify_partner_ids.ids
-                            )]
-                        })
-                        _logger.info(
-                            'Notifying partners %s about message with '
-                            'Message-Id: %s' % (
-                                fetchmail_server.e_inv_notify_partner_ids.ids,
-                                message.get('Message-Id')))
-                    else:
-                        _logger.error(
-                            'Can\'t notify anyone about not processed '
-                            'PEC E-Mail with Message-Id: {}'.format(
-                                message.get('Message-Id')))
-                return []
-
-        return super(MailThread, self).message_route(
-            message, message_dict, model=model, thread_id=thread_id,
-            custom_values=custom_values)
+                    return []
+                _logger.info(
+                    'Can\'t route PEC E-Mail with Message-Id: {}'.format(
+                        message.get('Message-Id'))
+                )
+                raise UserError(_(
+                    "PEC message with Message-Id %s has been read "
+                    "but not processed, as not related to an "
+                    "e-invoice.\n"
+                    "Please check PEC mailbox %s, at server %s,"
+                    " with user %s."
+                ) % (
+                    message.get('Message-Id'),
+                    fetchmail_server.name, fetchmail_server.server,
+                    fetchmail_server.user
+                ))
+        else:
```

Comment:
E se `self._context.get('fetchmail_server_id')` e `not fetchmail_server.is_fatturapa_pec`? Il messaggio viene perso? In questo modo non vengono perse tutte le email di mailbox normali?

Reply:
Giusto, mi sono perso un ramo dell'`if`!
Ho corretto e snellito un po' il metodo per renderlo più leggibile